infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -237,13 +244,11 @@ private void process() {
                     lastOp = currentOp;
                 }
             }
-        } catch (Exception e) {
-            log.error("Caught exception in the write processor ", e);
-            context.setErrorStatus();
         } catch (Throwable th) {
+            // Initiate a shutdown of the JVM when an unhandled Exception/Throwable is
+            // encountered, including IOExceptions during fsync and OutOfDirectMemoryErrors.
             log.error("Encountered throwable in the write processor ", th);
-            context.setErrorStatus();
-            throw th;
+            exitManager.exit(CorfuServer.EXIT_ERROR_CODE);
```

Comment:
I wonder if it's better to just call `Runtime.getRuntime().halt(CorfuServer.EXIT_ERROR_CODE)`  because `System.exit` would invoke the shutdown hook which might cause a deadlock if there exists some shutdown logic that is dependent on persistence.

Reply:
I thought about this too and specifically went with `System.exit()` to avoid the abrupt shutdown. I am concerned that by invoking halt() instead, the abrupt shutdown can cause DataCorruption on startup with unfortunate timing.